data_steward/gcloud/gcs/__init__.py

```diff
@@ -166,14 +170,17 @@ def _get_hpo_bucket_id(self, hpo_id: str) -> str:
         or (2) multiple records are found in the lookup table.
         """
         service = os.environ.get('GAE_SERVICE', 'default')
+        LOGGER.info(f"GAE_SERVICE: {service}")
 
         hpo_bucket_query = JINJA_ENV.from_string(SELECT_ALL_QUERY).render(
             project_id=self.project,
             dataset_id=LOOKUP_TABLES_DATASET_ID,
             table_id=HPO_ID_BUCKET_NAME_TABLE_ID)
 
+        LOGGER.info(f"Running: {hpo_bucket_query}")
+
         result_df: DataFrame = query(hpo_bucket_query)
-        condition_hpo_id = (result_df['hpo_id'] == hpo_id)
+        condition_hpo_id = ((result_df['hpo_id'].str.lower()) == hpo_id.lower())
         condition_service = (result_df['service'] == service)
         result_filtered = result_df[condition_hpo_id & condition_service]
```

Comment:
@ksdkalluri What is the advantage of selecting everything over what we are specifically looking for?  Wouldn't this work:

```suggestion
        # select query is:
        # select * from `{{project_id}}.{{dataset_id}}.{{table_id}}` where lower(hpo_id) = lower('{{hpo_id}}')
        hpo_bucket_query = JINJA_ENV.from_string(SELECT_QUERY).render(
            project_Id=self.project,
            dataset_id=LOOKUP_TABLES_DATASET_ID,
            table_id=HPO_ID_BUCKET_NAME_TABLE_ID,
            hpo_id=hpo_id)

        result_df: DataFrame = query(hpo_bucket_query)

        if result_df['bucket_name'].count() != 1:
            reutnr 'None'

        return result_df['bucket_name'].iloc[0]
```

To me, the code is shorter and easier to understand.  And I've also been in situations where calling "get all" works fine in test but leads to severe performance issues in production.

Reply:
That is a better approach, I will test it.